docs/generate_svgs.py

```diff
@@ -0,0 +1,45 @@
+import re
+import logging
+import subprocess
+
+from mkdocs.structure.pages import Page
+from mkdocs.structure.files import Files
+from mkdocs.config.defaults import MkDocsConfig
+
+
+logger = logging.getLogger('mkdocs')
+rzk_code_block = re.compile(r'(^```\s*rzk[^\n]*\s+(.*?)\s+^```)', flags=re.MULTILINE | re.DOTALL)
+svg_element = re.compile(r'^(<svg.*?</svg>)', flags=re.MULTILINE | re.DOTALL)
+rzk_installed = True
+
+try:
+    # Capture output to prevent logging usage
+    subprocess.run('rzk', capture_output=True)
+except FileNotFoundError:
+    logger.warn('Rzk executable not found')
+    rzk_installed = False
+
+
+def on_page_markdown(md: str, page: Page, config: MkDocsConfig, files: Files) -> str:
+    if not rzk_installed: return md
+    # Some snippets can depend on terms defined in previous snippets, so we need to store them all
+    previous_snippets = ['#lang rzk-1\n#set-option "render" = "svg"\n\n']
```

Comment:
This way you do not check if `#set-option` is called in the middle of the code, or if it is immediately unset. I would just ignore this and assume we always render something.

Reply:
Right, but if it is unset somewhere then we probably don't want to render a diagram for that snippet, no?

> I would just ignore this and assume we always render something

You mean ignore the clearing of the option (as in remove any `#set-option "render" = "none"` that might be in the snippet) or ignore the fact that it can be cleared?

Reply:
I misunderstood this line, keep as is, please.